src/types.rs

```diff
@@ -85,7 +85,16 @@ fn deserialize_semantic_tokens<'de, D>(
 where
     D: Deserializer<'de>,
 {
-    Vec::deserialize(deserializer).map_err(|e| {
```

Comment:
we could include the old deserializer as fallback

I get that we do want to tell users about the new syntax, but I suspect that most people don't even use semantic tokens, so they don't really care. Of course we make other improvements to `kak-lsp.toml`, so it's good practice to start fresh from the default config once in a while. Pressing the issue is not necessary, but we didn't have any complaints last time, so I doubt it matters. LGTM either way.

Reply:
Since the change is just a "cosmetic" syntax change, I think it would be fine to keep both.
Especially since "expressive names vs. concise definitions" is usually about personal preference, and we do have the ability to offer both.

Reply:
> `"" = "my_face"` will match everything.

you mean every token?
Or the entire buffer? In that case we should probably reject it.

Reply:
Every token individually.
Rejecting it would probably make sense either way, since there really is no benefit to it.

Reply:
I agree, rejecting the empty string makes sense